### PR TITLE
Clean up access to router state.

### PR DESCRIPTION
**Pull request by jwhui**

### CODE REVIEW

File: src/core/thread/mle_router.cpp [hunk 1]
```diff
@@ -1350,23 +1348,18 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
...
+            router->mLinkInfo.Clear();
+            router->mLinkInfo.AddRss(mMac.GetNoiseFloor(), threadMessageInfo->mRss);
+            router->mLinkFailures = 0;
+            router->mState = Neighbor::kStateLinkRequest;
+            SendLinkRequest(router);
+            ExitNow(error = kThreadError_NoRoute);
         }
 
-        router->mLastHeard = Timer::GetNow();
+        mParent.mLastHeard = Timer::GetNow();
```
Comment (librasungirl) [inline src/core/thread/mle_router.cpp]: should this line put into `if (memcmp(&mParent.mMacAddr, &macAddr, sizeof(mParent.mMacAddr)) == 0) `block?

File: src/core/thread/mle_router.cpp [hunk 2]
```diff
@@ -1042,26 +1046,23 @@ uint8_t MleRouter::LqiToCost(uint8_t aLqi)
 
 uint8_t MleRouter::GetLinkCost(uint8_t aRouterId)
 {
-    uint8_t rval;
+    uint8_t rval = kInvalidRouterId;
```
Comment (librasungirl) [inline src/core/thread/mle_router.cpp]: update `kInvalidRouterId` to `kMaxRouteCost`?

File: src/core/thread/mle_router.cpp [hunk 3]
```diff
@@ -1042,26 +1046,23 @@ uint8_t MleRouter::LqiToCost(uint8_t aLqi)
...
 
-    // kInvalidRouterId indicates non-existing next hop, hence return kMaxRouteCost for it.
-    assert(aRouterId <= kInvalidRouterId);
+    router = GetRouter(aRouterId);
 
-    VerifyOrExit(aRouterId != mRouterId &&
-                 IsRouterIdValid(aRouterId) &&
-                 mRouters[aRouterId].mState == Neighbor::kStateValid,
-                 rval = kMaxRouteCost);
+    // kInvalidRouterId indicates non-existing next hop, hence return kMaxRouteCost for it.
```
Comment (nibanks) [inline src/core/thread/mle_router.cpp]: Maybe change the `kInvalidRouterId indicates` comment to `NULL aRouterId indicated`.

File: src/core/thread/mle_router.cpp (hunk 1 above)
Comment (jwhui) [inline src/core/thread/mle_router.cpp]: Yes, thank!  Updated.

File: src/core/thread/mle_router.cpp (hunk 3 above)
Comment (jwhui) [inline src/core/thread/mle_router.cpp]: Yes, thanks.  Updated.

**Pull request merged**